.goreleaser.yaml

```diff
@@ -163,6 +163,16 @@ docker_manifests:
 checksum:
   name_template: "{{ .ProjectName }}_{{ .Version }}_SHA256SUMS"
 
+signs:
+  - artifacts: checksum
+    cmd: cosign
+    certificate: "${artifact}.pem"
+    args: ["sign-blob", "--output-certificate=${certificate}", "--output-signature=${signature}", "${artifact}", "--yes"]
```

Comment:
For the first iteration, please ensure that we are not using default settings:
--oidc-issuer string  OIDC provider to be used to issue ID token (default "https://oauth2.sigstore.dev/auth")
but instead, we are acting as a GitHub action as workload identity
--oidc-issuer=https://token.actions.githubusercontent.com

When the value is not defined probably, we are using the default one.

Reply:
Running the action via GHA makes cosign use the OIDC ID token of the GHA, and https://token.actions.githubusercontent.com/ as the issuer

But I'll make it explicit